src/Parser/Druid/Balance/Modules/Core/GlobalCooldown.js

```diff
@@ -0,0 +1,56 @@
+import SPELLS from 'common/SPELLS';
+import CoreGlobalCooldown from 'Parser/Core/Modules/GlobalCooldown';
+import Channeling from 'Parser/Core/Modules/Channeling';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Abilities from '../Features/Abilities';
+import Haste from './Haste';
+
+/**
+ * The talent Starlord reduces GCD and cast time of empowered Lunar Strikes and Solar Wraths by 20%. 
+ * Since Solar Wrath cast time == GCD the GCD needs to be reduced.
+ */
+class GlobalCooldown extends CoreGlobalCooldown {
+  static dependencies = {
+    abilities: Abilities,
+    haste: Haste,
+    combatants: Combatants,
+    channeling: Channeling,
+  };
+
+  isStarlordBuffedSolarWrath(event) {
+  	return event.ability.guid === SPELLS.SOLAR_WRATH_MOONKIN.id && this.combatants.selected.hasTalent(SPELLS.STARLORD_TALENT.id) && this.combatants.selected.hasBuff(SPELLS.SOLAR_EMP_BUFF.id);
+  }
+
+  on_byPlayer_begincast(event) {
+    if (this.isStarlordBuffedSolarWrath(event)) {
+      this.triggerStarlordGlobalCooldown(event, 'begincast');
+    } 
+    return;
+  }
+  on_byPlayer_beginchannel(event) {
+  	if (this.isStarlordBuffedSolarWrath(event)) {
+      return;
+    } 
+    super.on_byPlayer_beginchannel(event);
+  }
+  on_byPlayer_cast(event) {
+  	if (this.isStarlordBuffedSolarWrath(event)) {
+      return;
+    } 
+    super.on_byPlayer_cast(event);
+  }
+
+  triggerStarlordGlobalCooldown(event, trigger) {
+    this.owner.triggerEvent('globalcooldown', {
+      type: 'globalcooldown',
+      ability: event.ability,
+      timestamp: event.timestamp,
+      duration: Math.max(750, this.getCurrentGlobalCooldown(event.ability.guid) * 0.8), //Starlord multiplier
```

Comment:
Are you sure the minimum is 750 with the buff? Most of the time the minimum GCD is exactly 50% of the default, and this buff seems to reduce the *default* by 20%.

Reply:
yeah afaik the only way to reduce GCD below the minimum 750 is the Elisande buff from NH, atleast for balance. I think it's the same for all casters.